### PR TITLE
Laboratory work #3, Egorova Anna - 21FPL1

### PR DESCRIPTION
**Pull request by chezar0**

### CODE REVIEW

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -251,7 +308,15 @@ def is_incidental(self, vertex1: int, vertex2: int) -> int:
                 1 if vertices are incidental, otherwise 0
         If either of vertices is not present in the graph, -1 is returned
         """
-        pass
+        for vertex in vertex1, vertex2:
+            if vertex not in self._vertexes:
+                return -1
+        if self._matrix[self._vertexes.index(vertex1)][self._vertexes.index(vertex2)] == 1 and\
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:314]: split into two `ifs`

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py [hunk 2]
```diff
@@ -152,8 +176,13 @@ def decode(self, encoded_tokens: tuple[int, ...]) -> Optional[tuple[str, ...]]:
                 sequence of string tokens
         In case of out-of-dictionary input data, None is returned
         """
-        pass
-
+        for i in encoded_tokens:
+            if i not in self._id2word.keys():
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:180]:
```suggestion
            if i not in self._id2word:
```

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py [hunk 3]
```diff
@@ -36,7 +39,8 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
             punctuation : tuple[str, ...]
                 punctuation symbols to remove during text cleaning
         """
-        pass
+        self.stop_words = stop_words
```
[the remark this answers is not in the document]
Comment (chezar0) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py [hunk 4]
```diff
@@ -152,8 +176,13 @@ def decode(self, encoded_tokens: tuple[int, ...]) -> Optional[tuple[str, ...]]:
                 sequence of string tokens
         In case of out-of-dictionary input data, None is returned
         """
-        pass
-
+        for i in encoded_tokens:
+            if i not in self._id2word.keys():
+                return None
+        words = []
+        for i in encoded_tokens:
```
[the remark this answers is not in the document]
Comment (chezar0) [reply lab_3_keywords_textrank/main.py:179]: done

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (chezar0) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py (hunk 2 above)
Comment (chezar0) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py [hunk 5]
```diff
@@ -66,7 +73,11 @@ def _remove_stop_words(self, tokens: tuple[str, ...]) -> tuple[str, ...]:
             tuple[str, ...]
                 tokens without stop-words
         """
-        pass
+        new_tokens = []
+        for i in tokens:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:77]: do it in one line

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py (hunk 5 above)
Comment (chezar0) [reply lab_3_keywords_textrank/main.py]: done

**Review by artyomtugaryov — APPROVED** [review]

**Pull request closed without merging**